cmd/bootstrap/runner.go

```diff
@@ -511,3 +561,42 @@ func (r *runner) installOperator(ctx context.Context, helmClient helmclient.Inte
 
 	return nil
 }
+
+func (r *runner) patchChartOperatorDeployment(ctx context.Context, k8sClients k8sclient.Interface) error {
+	labelSelector := "app.kubernetes.io/instance=chart-operator-unique"
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waiting for chart-operator with label selector %#q", labelSelector))
+
+	o := func() error {
+		list, err := k8sClients.K8sClient().AppsV1().Deployments(namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+
+		if apierrors.IsNotFound(err) || len(list.Items) != 1 {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "chart-operator deployment not ready yet")
+			// fall through
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		d := list.Items[0]
+
+		d.Spec.Template.Spec.DNSConfig = nil
+		d.Spec.Template.Spec.DNSPolicy = corev1.DNSClusterFirst
+
+		_, err = k8sClients.K8sClient().AppsV1().Deployments(namespace).Update(ctx, &d, metav1.UpdateOptions{})
+
```

Comment:
Can you remove the newline here too?

This will work fine but you could also patch the deployment. Here is an example from chart-operator where we patch chart CR annotations.

https://github.com/giantswarm/chart-operator/blob/cd10b65ce47edcc73648b804d322bbf9cc9be704/service/controller/chart/resource/release/resource.go#L138-L166

Reply:
Sure, I'll use patching :)